sdk/java/pulumi/src/main/java/com/pulumi/core/Output.java

```diff
@@ -221,20 +224,39 @@ private static <T> Output<List<T>> all(List<Output<T>> outputs) {
      * Similarly, if any of the {@link Output}s are secrets,
      * then the final result will be a secret.
      *
-     * @param formattableString the format string with same syntax as expected by {@link String#format(String, Object...)}
-     * @param arguments         {@link Output}s that values of will be applied to the format String
+     * @param formattableString The format string with same syntax as expected by {@link String#format(String, Object...)},
+     *                          the behaviour on a {@code null} argument depends on the format conversion.
+     * @param arguments         {@link Output}s or regular {@link Object}s that values of will be applied to the format String
      * @return an {@link Output} with the result of the formatting
      */
-    static Output<String> format(String formattableString, @SuppressWarnings("rawtypes") Output... arguments) {
-        var data = Lists.newArrayList(arguments).stream()
+    static Output<String> format(String formattableString, @Nullable Object... arguments) {
```

Comment:
I wonder why does the implementation need to drop to OutputData layer?

You have `<T> Output<List<T>> all(List<Output<T>> outputs)`.

Why cannot that be used here? 

`List<Object> --> List<Output<Object>> --> Output<List<Object>> --> Output<String>`

Reply:
https://github.com/pulumi/pulumi-java/issues/552 is the bug

Reply:
Yes, that one was doing weird things, didn't want to mix the two issues.

Reply:
Yes, that makes sense. I think the implementation is functionally okay it does the same thing, just weird.

Reply:
as for the dropping, `.map(OutputData::copyInputOutputData)` this defensive copy looks like the only reason